pkg/daemon/daemon.go

```diff
@@ -742,3 +666,58 @@ func ValidPath(path string) bool {
 	}
 	return false
 }
+
+// SenseAndLoadOnceFrom gets a hold of the content for supported onceFrom configurations,
+// parses to verify the type, and returns back the genericInterface, the type description,
+// if it was local or remote, and error.
+func (dn *Daemon) SenseAndLoadOnceFrom() (interface{}, string, string, error) {
+	var content []byte
+	var err error
+	var contentFrom string
+	// Read the content from a remote endpoint if requested
+	if strings.HasPrefix(dn.onceFrom, "http://") || strings.HasPrefix(dn.onceFrom, "https://") {
+		contentFrom = MachineConfigOnceFromRemoteConfig
+		resp, err := http.Get(dn.onceFrom)
+		if err != nil {
+			return nil, "", contentFrom, err
+		}
+		defer resp.Body.Close()
+		// Read the body content from the request
+		content, err = dn.fileSystemClient.ReadAll(resp.Body)
+		if err != nil {
+			return nil, "", contentFrom, err
+		}
+	} else {
+		// Otherwise read it from a local file
+		contentFrom = MachineConfigOnceFromLocalConfig
+		absoluteOnceFrom, err := filepath.Abs(filepath.Clean(dn.onceFrom))
+		if err != nil {
+			return nil, "", contentFrom, err
+		}
+
+		content, err = dn.fileSystemClient.ReadFile(absoluteOnceFrom)
+		if err != nil {
+			return nil, "", contentFrom, err
+		}
+	}
+
+	// Try each supported parser
+	// TODO: Add to resource/machineconfig.go as a function?
+	mcfgScheme := runtime.NewScheme()
```

Comment:
We can move this logic and below to the calling function to avoid having to return an interface, just return the file contents.  Probably would make testing easier as well.